code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -88,21 +88,22 @@ Pipelines + Other Objects -> Pipe network
 		var/con_dir = get_dir(src, connected_node)
 		missing_nodes -= con_dir // finds all the directions that aren't pointed to by a node
 		var/image/nodecon = node_con["[con_dir]"]
-		if (default_colour && connected_node.default_colour && (connected_node.default_colour != default_colour)) // if both pipes have special colours - average them
-			var/list/centre_colour = GetHexColors(default_colour)
-			var/list/other_colour = GetHexColors(connected_node.default_colour)
-			var/list/average_colour = list(((centre_colour[1]+other_colour[1])/2),((centre_colour[2]+other_colour[2])/2),((centre_colour[3]+other_colour[3])/2))
-			nodecon.color = rgb(average_colour[1],average_colour[2],average_colour[3])
-		else if (color)
-			nodecon.color = null
-		else if (connected_node.color)
-			nodecon.color = connected_node.color
-		else if(default_colour)
-			nodecon.color = default_colour
-		else if(connected_node.default_colour && connected_node.default_colour != "#B4B4B4")
-			nodecon.color = connected_node.default_colour
-		else nodecon.color = "#B4B4B4"
-		underlays += nodecon
+		if(nodecon)

```

Comment:
Except it shouldn't be null in any cases, that points to a completely different problem that this doesn't address at all